sdk/core/azure-core/src/http/curl/curl_session_private.hpp

```diff
@@ -220,17 +215,17 @@ namespace Azure { namespace Core { namespace Http {
      * decide how much data to take from the inner buffer before pulling more data from network.
      *
      */
-    int64_t m_bodyStartInBuffer;
+    int64_t m_bodyStartInBuffer = -1;
```

Comment:
Out of curiosity, why do we start this at -1?

Reply:
This variable is used when after reading from the network, we have read and recorded part of the body into our cache (inner body). When that happen, this variable tell where is the start position of the body.
When it is -1, it means there is not  a body-content in the cache. We can't use 0 because that would indicate that there the body starts at position 0.